src/python/pants/backend/python/tasks/setup_py.py

```diff
@@ -672,7 +667,7 @@ def create(exported_python_target):
                     exported_python_target, exported_python_target
                 )
                 setup_dir, dependencies = self.create_setup_py(subject, dist_dir)
-                created[exported_python_target] = setup_dir
+                created[exported_python_target] = Path(setup_dir)
```

Comment:
Ditto on if this can use `PurePath`.

Reply:
Same answer / same reason - this gets shuffled to SetupPyRunner which needs a Path to check `.is_dir()`.